mqtt/mqtt-bridge/src/persist/memory/loader.rs

```diff
@@ -0,0 +1,431 @@
+use std::{pin::Pin, sync::Arc, task::Context, task::Poll, vec::IntoIter};
+
+use futures_util::stream::Stream;
+use mqtt3::proto::Publication;
+use parking_lot::{Mutex, MutexGuard};
+
+use crate::persist::{memory::waking_map::WakingMap, Key};
+
+// Message loader used to extract elements from bridge queues
+// This component is responsible for message extraction from the queue
+// It works by grabbing a snapshot of the most important messages from the queue
+// Then, will return these elements in order
+// When the batch is exhausted it will grab a new batch
+// If no elements are removed from the queue, the second batch will be identical to first
+pub struct InMemoryMessageLoader {
```

Comment:
thanks for the code docs, they really help. Small nit: use /// for [code docs](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html).